benchexec/tools/condtesttools.py

```diff
@@ -0,0 +1,46 @@
+"""
+BenchExec is a framework for reliable benchmarking.
+This file is part of BenchExec.
+
+Copyright (C) 2007-2017  Dirk Beyer
+All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import benchexec.tools.template
+import benchexec.result as result
+import benchexec.util as util
+
+
+class Tool(benchexec.tools.template.BaseTool):
+    """
+    An abstract class for the conditonal testing tools.
+    """
+
+    def executable(self):
+        return util.find_executable(self._exec_path)
+
+    def version(self, executable):
+        return self._version_from_tool(executable)
+
+    def cmdline(self, executable, options, tasks, propertyfile, rlimits):
+        return (
+            [executable]
+            + options
+            + ["--spec"]
+            + [propertyfile or "None"]
+            + [f for f in tasks]
+        )
+
+    def determine_result(self, returncode, returnsignal, output, isTimeout):
```

Comment:
Overriding this method is not necessary because its default implementation is the same.